packages/amplify_flutter/lib/amplify.dart

```diff
@@ -184,6 +186,13 @@ class AmplifyClass extends PlatformInterface {
     await DataStore.configure(configuration);
   }
 
+  void _castAndThrowAnalyticsException(PlatformException e) {
+    switch (e.code) {
+      case "AnalyticsException": throw AnalyticsException.fromMap(Map<String, String>.from(e.details));
+      default: throw e;
+    }
+  }
```

Comment:
We can just throw `AnalyticsException.fromMap(Map<String, String>.from(e.details));` inline